src/utils/lupaParser.js

```diff
@@ -1,35 +1,94 @@
 import {
-  KOHTEET,
-  LUPA_SECTIONS
+  GENERIC_LUPA_SECTIONS, VST_LUPA_STRUCTURE
 } from "../scenes/VapaaSivistystyo/modules/constants";
 import { parseLocalizedField } from "../modules/helpers";
+import common from "../i18n/definitions/common";
 
 /**
- * Return an object with lupamääräys data categorized into sections as required in Kuja
+ * Return an object with lupamääräys data categorized into sections matching VST template in
+ * kuja-template/lupahistoria/liikunnankoulutuskeskukset/paatos
+ *
+ * Depends on VST_LUPA_STRUCTURE to guide the order and logic of parsing määräys data and
+ * producting formatted section data, based on kohde or koodisto identifier. Uses react-intl
+ * utilities for primary localization and that based on koodisto data.
+ *
+ * @param lupa lupa with määräys and järjestäjä as given by backend
+ * @param intl react-intl utility object
+ * @returns [{heading: string, values: string[]}]
+ */
+export const parseVSTLupa = (lupa, intl) => {
+  if(lupa) {
+    const organizerSectionData = generateOrganizerSectionData(lupa, intl.locale);
+    organizerSectionData.heading = intl.formatMessage(common.VSTLupaSectionTitleOrganizer)
+    const sectionDataList = [
+      organizerSectionData
+    ];
+    for (const metaDataObject of VST_LUPA_STRUCTURE) {
+      let maaraykset = [];
+      let generateSectionData = () => ({});
+      if(metaDataObject.kohdeTunniste) {
+        maaraykset = lupa.maaraykset.filter(maarays => maarays.kohde.tunniste === metaDataObject.kohdeTunniste);
+        generateSectionData = getSectionDataGeneratorForVST(metaDataObject.kohdeTunniste);
+      }
+      else if(metaDataObject.koodisto) {
+        generateSectionData = getSectionDataGeneratorForVST(metaDataObject.koodisto);
+        maaraykset = lupa.maaraykset.filter(maarays => maarays.koodisto === metaDataObject.koodisto);
+      }
+
+      let sectionData = {};
+      if(maaraykset.length > 0)  {
+        sectionData = generateSectionData(maaraykset, intl.locale);
+      }
+      sectionData.heading = intl.formatMessage(metaDataObject.titleMessageKey);
+      sectionDataList.push(sectionData)
+
+    }
+    return sectionDataList;
+  }
+};
+
+const generateOrganizerSectionData = (lupa, locale) => {
+  const kunta = parseLocalizedField(lupa.jarjestaja.kuntaKoodi.metadata, locale.toUpperCase())
+  // Exception sourced from kuja-template/lupahistoria/liikunnankoulutuskeskukset/paatos/content_paatos_fi.html:35
+  // TODO: localization of this exception case content
+  const value = lupa.jarjestaja.oid === '1.2.246.562.10.13451568789' ?
+    `${lupa.jarjestaja.nimi[locale]}, ${kunta} sekä Humppilan ja Ypäjän kunnat` :
```

Comment:
Swedish translation and localization in common not needed?

Reply:
Same kind of hardcoded exception in the html template. Should be fixed separately and out of scope here.